tests/func-tests/virtual_machines_test.go

```diff
@@ -0,0 +1,56 @@
+package tests
+
+import (
+	"flag"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	k8smetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/rand"
+
+	kubevirtv1 "kubevirt.io/client-go/api/v1"
+	"kubevirt.io/client-go/kubecli"
+)
+
+const timeout = 240 * time.Second
+const pollingInterval = 5 * time.Second
+
+var _ = Describe("Virtual Machines", func() {
+
+	kubecli.Init()
+	flag.Parse()
+	client, err := kubecli.GetKubevirtClient()
+	if err != nil {
+		panic(err)
+	}
+
+	var vmi *kubevirtv1.VirtualMachineInstance
+	var vmiRandName = vmiName + rand.String(48)
+	vmi = kubevirtv1.NewMinimalVMIWithNS(testNamespace, vmiRandName)
+
+	Context("vmi testing", func() {
+		It("should create a VirtualMachineInstance", func() {
+			Eventually(func() error {
+				_, err := client.VirtualMachineInstance(testNamespace).Create(vmi)
+				return err
+			}, timeout, pollingInterval).Should(Not(HaveOccurred()), "failed to create a vmi")
+		})
+
+		It("should verify vmi Pashe is Running", func() {
+			Eventually(func() bool {
+				vmi, err = client.VirtualMachineInstance(testNamespace).Get(vmiRandName, &k8smetav1.GetOptions{})
```

Comment:
each individual test should be self contained and not pend on other test cases. For example, this test that waits for vmi.Status.Phase == Running is dependent on that VMI getting created in another test. Instead, this test should both create the VMI, ensure that it reaches the Running Phase, and then delete the VMI.

There only needs to be a single It() defined for this